File: c_example.R
library( ANTsR )

# Read in the images

dataDirectory <- './data/'

fixedImage <- antsImageRead(
   paste0( dataDirectory, 'chalf.nii.gz' ), dimension = 2 )
movingImage <- antsImageRead(
   paste0( dataDirectory, 'c.nii.gz' ), dimension = 2 )

# Plot the fixed  and moving images
plot( movingImage, fixedImage, color.overlay = "jet", alpha = 0.7 )

#######
#
# Perform registration
#

outputDirectory <- './OutputANTsR/'
if( ! dir.exists( outputDirectory ) )
  {
  dir.create( outputDirectory )
  }
outputPrefix <- paste0( outputDirectory, 'antsr' )

registration <- antsRegistration(
  fixed = fixedImage, moving = movingImage,
  typeofTransform = 'TVMSQC',
  gradStep = 2.0,
  verbose = TRUE, printArgs = TRUE, outprefix = outputPrefix )

antsImageWrite( registration$warpedmovout, paste0( outputPrefix, "Warped.nii.gz" ) )
antsImageWrite( registration$warpedfixout, paste0( outputPrefix, "InverseWarped.nii.gz" ) )

# Plot the moving and warped fixed image
plot( movingImage, registration$warpedfixout, color.overlay = "jet", alpha = 0.4 )

File: c_example.py
import ants
import os

# Read in the images

dataDirectory = './data/'

fixedImageFile = dataDirectory + 'chalf.nii.gz'
fixedImage = ants.image_read( fixedImageFile, dimension = 2 )
movingImageFile = dataDirectory + 'c.nii.gz'
movingImage = ants.image_read( movingImageFile, dimension = 2 )

# Plot the fixed  and moving images
ants.plot( movingImage, overlay = fixedImage, overlay_cmap = "viridis", alpha = 0.9 )

#######
#
# Perform registration
#

outputDirectory = './OutputANTsPy/'
if not os.path.isdir( outputDirectory ):
  os.mkdir( outputDirectory )

outputPrefix = outputDirectory + 'antsr'

registration = ants.registration(
  fixed = fixedImage, moving = movingImage,
  type_of_transform = 'TVMSQC',
  grad_step = 2.0,
  verbose = True, outprefix = outputPrefix )

ants.image_write( registration['warpedmovout'], outputPrefix + "Warped.nii.gz" )
ants.image_write( registration['warpedfixout'], outputPrefix + "InverseWarped.nii.gz" )

# Plot the moving and warped fixed image
ants.plot( movingImage, overlay = registration['warpedfixout'], overlay_cmap = "viridis", alpha = 0.9 )
